### PR TITLE
Next round of X.org libraries

### PR DESCRIPTION
**Pull request by pkgw**
- libICE
- libSM
- libpthread-stubs
- xcb-proto
- xtrans

These are a bit more complex to build than the proto packages but I've checked out their internals and I think I have their dependencies right.

### CODE REVIEW

**Comment by pkgw** [comment]
Kicking CI; key AppVeyor build seems not to have gotten registered.

**Comment by pkgw** [comment]
Finally! We laughed, we cried, we learned. Should be good to merge.

File: recipes/xorg-xcb-proto/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+build:
+  number: 0
+  detect_binary_files_with_prefix: true
+
+requirements:
+  build:
+    - m2w64-pkg-config  # [win]
+    - pkg-config  # [not win]
+    - posix  # [win]
+    - python >=2.7,<3
```
Comment (scopatz) [inline recipes/xorg-xcb-proto/meta.yaml]: Similar question. Can we add skips for these versions of Python. Also, does it not build with Python 3?

**Comment by scopatz** [comment]
Impressive. Just the Python version question

File: recipes/xorg-xcb-proto/meta.yaml (hunk 1 above)
Comment (pkgw) [inline recipes/xorg-xcb-proto/meta.yaml]: The Python code does not appear to be compatible with Python 3 — `print` is used as a keyword in `xcbgen/xtypes.py`.

As with `gobject-introspection`, I'm pretty sure that the builders don't matrix over the Python versions as-is, so I don't think a versioned skip would be helpful? If I'm understanding your question correctly.

**Comment by pkgw** [comment]
CI finally finished; should be good to merge.

**Comment by scopatz** [comment]
Thanks @pkgw!

File: recipes/xorg-xcb-proto/meta.yaml (hunk 1 above)
Comment (ocefpaf) [inline recipes/xorg-xcb-proto/meta.yaml]: With the skip statement above you don't really need this here. But it does not hurt to leave it there.

**Pull request merged**